pkg/restore/restore.go

```diff
@@ -474,14 +485,55 @@ func (ctx *context) restoreFromDir(dir string) (api.RestoreResult, api.RestoreRe
 			if !existingNamespaces.Has(mappedNsName) {
 				logger := ctx.log.WithField("namespace", nsName)
 				ns := getNamespace(logger, filepath.Join(dir, api.ResourcesDir, "namespaces", api.ClusterScopedDir, nsName+".json"), mappedNsName)
-				if _, err := kube.EnsureNamespaceExists(ns, ctx.namespaceClient); err != nil {
+
+				clusterNS, err := ctx.namespaceClient.Get(mappedNsName, metav1.GetOptions{})
+				if err != nil && !apierrors.IsNotFound(err) {
 					addArkError(&errs, err)
 					continue
+				} else if apierrors.IsNotFound(err) {
+					_, err := ctx.namespaceClient.Create(ns)
+					if err != nil {
+						addArkError(&errs, err)
+						continue
+					}
+				}
+
+				// Namespace exists and is active
+				if clusterNS.Status.Phase == v1.NamespaceActive {
+					// keep track of namespaces that we know exist so we don't
+					// have to try to create them multiple times
+					existingNamespaces.Insert(mappedNsName)
```

Comment:
i would move the insertion into `existingNamespaces` down to below your `for` loop; it won't need to be within an `if` statement there - we'll always execute it if we get there.  See next comment as well.